clang/lib/DPCT/RulesAsm/AsmMigration.cpp

```diff
@@ -1093,6 +1093,132 @@ class SYCLGen : public SYCLGenBase {
     return HandleAddSub(Inst);
   }
 
+  bool HandleShfulSyncOp(const InlineAsmInstruction *Inst) {
+
+    if (emitStmt(Inst->getOutputOperand()))
+      return SYCLGenError();
+    OS() << " = ";
+
+    if (DpctGlobalInfo::useMaskedSubGroupFunction()) {
+
+      std::string Op[4];
```

Comment:
Operands generation can move out from the if branch. Because we do the same thing in else branch.

Reply:
Good catch, will update it.